### PR TITLE
HPCC-25475 Address Coverity scan results in Couchbase plugin

### PR DESCRIPTION
**Pull request by dcamper**
In addition to this PR, we should update the two submodules used by this plugin to point to the most current master branch commit.  The submodule's repos were forked by hpcc-systems:

https://github.com/hpcc-systems/libcouchbase.git
https://github.com/hpcc-systems/libcouchbase-cxx.git

Signed-off-by: Dan S. Camper <dan.camper@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Via regression tests.

### CODE REVIEW

**Comment by dcamper** [comment]
FYI:  Two Coverity errors for out-of-bounds access were declared false positives.

**Comment by dcamper** [comment]
Also: I recommend that the Coverity findings for the submodules be ignored for now, until we update our submodules to newer code.  It is probable that some or all of those findings have been addressed with newer code.

**Comment by dcamper** [comment]
I should add that these changes were successfully tested against Couchbase Server 7.0.0 (beta) and with both the existing submodules and with updated submodules.

File: plugins/couchbase/couchbaseembed.cpp [hunk 1]
```diff
@@ -912,32 +916,33 @@ namespace couchbaseembed
 
     IPropertyTree * CouchbaseEmbedFunctionContext::nextResultRowTree()
     {
-        for (auto cbrow : *m_pQuery)
+        if (m_pQuery)
         {
-            auto json = cbrow.json().to_string();
+            reportIfQueryFailure(m_pQuery);
```
Comment (rpastrana) [inline plugins/couchbase/couchbaseembed.cpp:921]: this makes sense. Per our offline conv, I'm wondering if reportIfQueryFailure should fail if m_pQuery contains multiple queries?
I'm not sure how often we'll encounter that situation, but I'm concerned about silently dropping the other queries.

**Review by rpastrana — COMMENTED** [review]
@dcamper looks good, just one comment which might not apply.

File: plugins/couchbase/couchbaseembed.cpp (hunk 1 above)
Comment (dcamper) [reply plugins/couchbase/couchbaseembed.cpp:921]: Well, with the original code, we never supported multiple queries anyway.  If we change anything, I would suggest only documenting this as a known limitation and then address it if the need arises.

Comment (rpastrana) [reply plugins/couchbase/couchbaseembed.cpp:921]: Sounds reasonable

**Review by rpastrana — DISMISSED** [review]

**Comment by dcamper** [comment]
@rpastrana Please check my README addendum.

**Review by rpastrana — APPROVED** [review]
@dcamper looks reasonable for now

**Pull request merged**